### PR TITLE
add util func can print by line.

### LINKED ISSUE #1422: Edgecore New configuration files | Info template
**What happened**:
Edgecore not running. Better output required.

**What you expected to happen**:
edgecore running

**How to reproduce it (as minimally and precisely as possible)**:
Build edgecore from the master 

**Anything else we need to know?**:
Currently, on running the edgecore with the new configuration files results in 
```
sudo ./edgecore
2020-01-21 02:04:52.445786 I | INFO: Install client plugin, protocol: rest
2020-01-21 02:04:52.446132 I | INFO: Installed service discovery plugin: edge
[TLSPrivateKeyFile: Invalid value: "/etc/kubeedge/certs/edge.key": TLSPrivateKeyFile not exist TLSCertFile: Invalid value: "/etc/kubeedge/certs/edge.crt": TLSCertFile not exist TLSCAFile: Invalid value: "/etc/kubeedge/ca/rootCA.crt": TLSCAFile not exist enable: Invalid value: false: websocket.enable and quic.enable cannot be true and false at the same time]
```
It would be suggested to have a newline between info messages. like

```
sudo ./edgecore
2020-01-21 02:04:52.445786 I | INFO: Install client plugin, protocol: rest
2020-01-21 02:04:52.446132 I | INFO: Installed service discovery plugin: edge
[
TLSPrivateKeyFile: Invalid value: "/etc/kubeedge/certs/edge.key": TLSPrivateKeyFile not exist
TLSCertFile: Invalid value: "/etc/kubeedge/certs/edge.crt": TLSCertFile not exist 
TLSCAFile: Invalid value: "/etc/kubeedge/ca/rootCA.crt": TLSCAFile not exist 
]
```
**Environment**:
Latest Master build

### PR DESCRIPTION
**Pull request by stacktasec**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>

> /kind bug

**What this PR does / why we need it**:
According to the suggest of issue #1422  add a func that can print the msg by line no matter it's collection or not, it help us read the msg types like 'errs' more easier.

**Which issue(s) this PR fixes**:

Fixes #1422

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: cloud/cmd/cloudcore/app/server.go [hunk 1]
```diff
@@ -2,10 +2,10 @@ package app
 
 import (
 	"fmt"
+	"github.com/kubeedge/kubeedge/pkg/util"
```
Comment (fisherxu) [inline cloud/cmd/cloudcore/app/server.go]: Move this line to the category of kubeedge below L14 :)

File: edge/cmd/edgecore/app/server.go [hunk 2]
```diff
@@ -3,11 +3,11 @@ package app
 import (
 	"errors"
 	"fmt"
+	"github.com/kubeedge/kubeedge/pkg/util"
```
Comment (fisherxu) [inline edge/cmd/edgecore/app/server.go]: ditto

File: keadm/cmd/keadm/keadm.go [hunk 3]
```diff
@@ -17,15 +17,15 @@ limitations under the License.
 package main
 
 import (
-	"fmt"
+	"github.com/kubeedge/kubeedge/pkg/util"
```
Comment (fisherxu) [inline keadm/cmd/keadm/keadm.go]: ditto

File: pkg/util/util.go [hunk 4]
```diff
@@ -514,3 +515,29 @@ func ReadDirNoStat(dirname string) ([]string, error) {
 
 	return f.Readdirnames(-1)
 }
+
+func PrintByLine(w io.Writer, msg interface{}) {
```
Comment (fisherxu) [inline pkg/util/util.go]:
```suggestion
func PrintByLine(w io.Writer, errors interface{}) {
```
Could you please add a unit test here? :)

**Comment by stacktasec** [comment]
@fisherxu I have fix the sort of import mentioned, and add an unit test.

File: edge/cmd/edgecore/app/server.go [hunk 5]
```diff
@@ -54,18 +54,18 @@ offering HTTP client capabilities to components of cloud to reach HTTP servers r
 			flag.PrintFlags(cmd.Flags())
 
 			if errs := opts.Validate(); len(errs) > 0 {
-				fmt.Fprintf(os.Stderr, "%v\n", utilerrors.NewAggregate(errs))
+				util.PrintByLine(os.Stderr, errs)
```
Comment (kadisi) [inline edge/cmd/edgecore/app/server.go]: we can use `klog.Fatal` like k8s: https://github.com/kubernetes/kubernetes/blob/fdb2cb4c8832da1499069bda918c014762d8ac05/cmd/kubelet/app/server.go#L164

Comment (fisherxu) [reply edge/cmd/edgecore/app/server.go]: great, so the PrintByLine only splice the errors

File: pkg/util/uitl_test.go [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+	"strings"
+	"testing"
+)
+
+func TestPrintByLine(t *testing.T) {
+	old := os.Stderr
+	r, w, _ := os.Pipe()
+	os.Stderr = w
+
+	errs := []error{
```
Comment (chendave) [inline pkg/util/uitl_test.go]: we'd better to consider other cases, map, default etc.

File: pkg/util/uitl_test.go [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+	if indexErr3 != len(head)+len(errLine1)+len(errLine2) {
+		t.Error("The func format the msg unexpected.")
+		return
+	}
+
+	indexTail := strings.Index(out, tail)
+	if indexTail != len(head)+len(errLine1)+len(errLine2)+len(errLine3) {
+		t.Error("The func format the msg unexpected.")
+		return
+	}
```
Comment (chendave) [inline pkg/util/uitl_test.go]: IMHO, too verbose for one case here.

**Comment by stacktasec** [comment]
@chendave I have added more case for unit test.

**Comment by stacktasec** [comment]
Emm, I think I should implement it by 'json.MarshalIndent'. Is it  better?

**Comment by stacktasec** [comment]
@fisherxu I have change the func name as you suggested, and rewrite unit test，and I merge the latest master branch to my branch, but I met a CI error 

> Device Management test in E2E scenario
/home/travis/gopath/src/github.com/kubeedge/kubeedge/tests/e2e/deployment/device_crd_test.go:41
  Test Device Instance Creation, Updation and Deletion
  /home/travis/gopath/src/github.com/kubeedge/kubeedge/tests/e2e/deployment/device_crd_test.go:157
    E2E_UPDATE_DEVICE_3: Update device instance for modbus protocol [It]
    /home/travis/gopath/src/github.com/kubeedge/kubeedge/tests/e2e/deployment/device_crd_test.go:397
    Expected
        <bool>: false
    to equal
        <bool>: true


What I write is only for printing msg, so why I meet this CI error?

File: pkg/util/util.go [hunk 8]
```diff
@@ -514,3 +515,37 @@ func ReadDirNoStat(dirname string) ([]string, error) {
...
+	}
+
+	var stb strings.Builder
+	stb.WriteString("[\n")
+	for _, errString := range errStrings {
+		stb.WriteString(fmt.Sprintf("  %s\n", errString))
+	}
+	stb.WriteString("]\n")
+	return stb.String()
+}
```
Comment (fisherxu) [inline pkg/util/util.go]:
```suggestion
func SpliceErrors(errors []error) string {
	if len(errStrings) == 0 {
		return ""
	}

	var stb strings.Builder
	stb.WriteString("[\n")
	for _, err := range errors {
		stb.WriteString(fmt.Sprintf("  %s\n", err.Error()))
	}
	stb.WriteString("]\n")
	return stb.String()
}
```
The errors in the parameter can only be slice.

**Comment by fisherxu** [comment]
> What I write is only for printing msg, so why I meet this CI error?

it's a flake test, let's resolve it later...

File: cloud/cmd/cloudcore/app/server.go [hunk 9]
```diff
@@ -40,19 +39,16 @@ kubernetes controller which manages devices so that the device metadata/status d
...
-				fmt.Fprintf(os.Stderr, "%v\n", err)
-				os.Exit(1)
+				klog.Fatal(err)
 			}
 
-			if errs := validation.ValidateCloudCoreConfiguration(config); len(errs) > 0 {
-				fmt.Fprintf(os.Stderr, "%v\n", errs)
-				os.Exit(1)
+			if errs := validation.ValidateCloudCoreConfiguration(config).ToAggregate().Errors(); len(errs) > 0 {
+				klog.Fatal(util.SpliceErrors(errs))
```
Comment (fisherxu) [inline cloud/cmd/cloudcore/app/server.go]:
```suggestion
			if errs := validation.ValidateCloudCoreConfiguration(config); len(errs) > 0 {
				klog.Fatal(util.SpliceErrors(errs.ToAggregate().Errors()))
```
avoiding the panic :)

**Comment by fisherxu** [comment]
/lgtm
Good job @stacktasec  ;-)

**Pull request merged**